content/articles/multi-output-classification-with-machine-learning/index.md

```diff
@@ -0,0 +1,315 @@
+from sklearn.multioutput import MultiOutputClassifier
+```
+Let's explain these packages that we have imported.
+
+**LogisticRegression**
+This is the algorithm used to train the model. 
+
+**CountVectorizer**
+Since we are dealing with text, we need to convert the input text into vectors of numbers. Machine learning models do not understand the raw text. The converted vectors of numbers are the representation of the original text. CountVectorizer is the most common Python package that is used to perform this process.
+
+For further reading on CountVectorizer and how they convert raw text into vectors of numbers, click [here](https://towardsdatascience.com/basics-of-countvectorizer-e26677900f9c)
+
+**train_test_split**
+This is the Python package that is used for dataset splitting. In machine learning, it's essential to split a dataset into two sets. One set is to be used for training and another one for testing.
+
+**accuracy_score**
+This is used to calculate the accuracy score of the model after training.
+
+**MultiOutputClassifier**
+Since we are dealing with a multi-output classification problem, we need a more specific algorithm.
+ `MultiOutputClassifier` is the most common [Scikit-learn](https://scikit-learn.org/stable/) algorithm that is used to build this model.
+
+We now need to specify features and labels for our model.
+
+### Adding features and labels
+
+Features and labels are essential in any machine learning label. Features represent all the columns used by the model as inputs during training. Labels represent the output or target columns, this is what the model wants to predict. We add the using the following code:
+
+```python
+Xfeatures = df['title']
+ylabels = df[['type','rating']]
+```
+From this code, our feature is the `title`, this is the input for our model. The labels are `type` and `rating`, these are the output for our model. We have two labels because we are dealing with a multi-output classification problem.
+
+The next step is to split our dataset using the `train_test_split` method.
+
+### Dataset splitting
+
+To split the dataset into two, use this code:
+
+```python
+x_train,x_test,y_train,y_test = train_test_split(Xfeatures,ylabels,test_size=0.3,random_state=7)
+```
+In the code above, we use a `test_size=0.3`. This will split our dataset so that `70%` of the dataset is used for training and `30%` will be used for testing. We have split our dataset, we are now ready to build the model.
+
+To build this model we will use the machine learning pipeline package to speed up the process of building our model. It will speed up the process by automating all the processes involved in building the model.
+
+The machine learning pipeline will automate the process of `CountVectorizer`. It will also automate the process of model training using `LogisticRegression` and `MultiOutputClassifier` algorithms.
+
+To implement this pipeline process, we will import the `Pipeline` package. 
+
+### Importing pipeline
+
```

Comment:
Consider removing all the white spaces after the heading. It's unnecessary as spaces are added automatically